src/main/java/teammates/ui/controller/Action.java

```diff
@@ -238,10 +238,9 @@ protected AccountAttributes authenticateAndGetNominalUser(UserType loggedInUserT
                 
                 throw new UnauthorizedAccessException("Unregistered user for a page that needs one");
             } else if (isPageNotCourseJoinRelated() && doesRegkeyBelongToUnregisteredStudent() && isUserLoggedIn) {
-                Url redirectUrl = new Url(Const.ViewURIs.LOGOUT)
-                                      .withParam(Const.ParamsNames.NEXT_URL, requestUrl)
-                                      .withUserId(StringHelper.encrypt(account.googleId)); 
-                
+                Url redirectUrl = new Url(student.getRegistrationUrl())
+                                      .withParam(Const.ParamsNames.STUDENT_NEW_COURSE, student.course)
+                                      .withParam(Const.ParamsNames.NEXT_URL, requestUrl);
```

Comment:
Hey, I would like to point out that (3) is not always true as the user *CAN* be loggedin without a course (although highly unlikely)